packages/core/test/acceptance/di_spec.ts

```diff
@@ -3546,4 +3546,599 @@ describe('di', () => {
+    const token = new InjectionToken<string>('greeting');
+
+    @Directive({selector: 'menu-trigger'})
+    class MenuTrigger {
+      @Input('triggerFor') menu!: TemplateRef<unknown>;
+
+      constructor(private viewContainerRef: ViewContainerRef) {}
+
+      open(injector: Injector|undefined) {
+        this.viewContainerRef.createEmbeddedView(this.menu, undefined, {injector});
+      }
+    }
+
+    it('should be able to provide an injection token through a custom injector', () => {
+      @Directive({selector: 'menu'})
+      class Menu {
+        constructor(@Inject(token) public tokenValue: string) {}
+      }
+
+      @Component({
+        template: `
+          <menu-trigger [triggerFor]="menuTemplate"></menu-trigger>
+          <ng-template #menuTemplate>
+            <menu></menu>
+          </ng-template>
+      `
+      })
+      class App {
+        @ViewChild(MenuTrigger) trigger!: MenuTrigger;
+        @ViewChild(Menu) menu!: Menu;
+      }
+
+      TestBed.configureTestingModule({declarations: [App, MenuTrigger, Menu]});
+      const injector = Injector.create({providers: [{provide: token, useValue: 'hello'}]});
+      const fixture = TestBed.createComponent(App);
+      fixture.detectChanges();
+
+      fixture.componentInstance.trigger.open(injector);
+      fixture.detectChanges();
+
+      expect(fixture.componentInstance.menu.tokenValue).toBe('hello');
+    });
+
+    it('should be able to provide an injection token to a nested template through a custom injector',
+       () => {
+         @Directive({selector: 'menu'})
+         class Menu {
+           constructor(@Inject(token) public tokenValue: string) {}
+         }
+
+         @Component({
+           template: `
+            <menu-trigger #outerTrigger [triggerFor]="outerTemplate"></menu-trigger>
+            <ng-template #outerTemplate>
+              <menu></menu>
+
+              <menu-trigger #innerTrigger [triggerFor]="innerTemplate"></menu-trigger>
+              <ng-template #innerTemplate>
+                <menu #innerMenu></menu>
+              </ng-template>
+            </ng-template>
+          `
+         })
+         class App {
+           @ViewChild('outerTrigger', {read: MenuTrigger}) outerTrigger!: MenuTrigger;
+           @ViewChild('innerTrigger', {read: MenuTrigger}) innerTrigger!: MenuTrigger;
+           @ViewChild('innerMenu', {read: Menu}) innerMenu!: Menu;
+         }
+
+         TestBed.configureTestingModule({declarations: [App, MenuTrigger, Menu]});
+         const fixture = TestBed.createComponent(App);
+         fixture.detectChanges();
+
+         fixture.componentInstance.outerTrigger.open(
+             Injector.create({providers: [{provide: token, useValue: 'hello'}]}));
+         fixture.detectChanges();
+
+         fixture.componentInstance.innerTrigger.open(undefined);
+         fixture.detectChanges();
+
+         expect(fixture.componentInstance.innerMenu.tokenValue).toBe('hello');
```

Comment:
nit: can we create a separate consts for the `fixture.componentInstance` fields, so the code is a bit more readable:
```suggestion
         const {outerTrigger, innerTrigger, innerMenu} = fixture.componentInstance;
         outerTrigger.open(
             Injector.create({providers: [{provide: token, useValue: 'hello'}]}));
         fixture.detectChanges();

         innerTrigger.open(undefined);
         fixture.detectChanges();

         expect(innerMenu.tokenValue).toBe('hello');
```

Reply:
The problem is that `innerTrigger` and `innerMenu` don't exist until `outerTrigger.open` is called.